src/PostgREST/App.hs

```diff
@@ -285,6 +305,15 @@ responseContentTypeOrError accepts action = serves contentTypesForRequest accept
             "None of these Content-Types are available: " <> failed
         Just ct -> Right ct
 
+binaryField :: ContentType -> [FieldName] -> Either Response (Maybe FieldName)
+binaryField CTOctetStream fieldNames = 
+  if length fieldNames == 1 && fromJust fieldName /= "*"
```

Comment:
What about `fieldName /= Just "*"`? I know in this case they are logically equivalent but I like to avoid partial functions like `fromJust`.